### DISCUSSION TITLE
how to read json file

### DISCUSSION THREAD

**Discussion by wonb168** [duckdb/duckdb#5679] [Ideas]
pandas can read_json file or json str to a df, like:
```
import pandas as pd 
j='[{"col 1":"a","col 2":"b"},{"col 1":"c","col 2":"d"}]'
df=pd.read_json(j, orient='records')
print(df)
```
does duckdb can do this? thanks!
```
import duckdb as dd 
j='[{"col 1":"a","col 2":"b"},{"col 1":"c","col 2":"d"}]'
df=dd.read_json(j, orient='records')
print(df)
```

**Comment by Mause** [comment]
This is being implemented in https://github.com/duckdb/duckdb/pull/5544 😁

Though you can already read json lines - depending on what exactly you're reading, you may be already able to load it

See https://duckdb.org/docs/extensions/json#json-creation-functions for more

**Reply by wonb168** [reply]
but this can only read json file?
how to read a json str, I use python requests get the json str.

and read_json_objects the result is wrong:
```
df=con.execute("select * from read_json_objects('d:/download/json.json')").df()
print(df)
json
0  [{"col 1":"a","col 2":"b"},{"col 1":"c","col 2...
```
pandas read_json got:
```
col 1 col 2
0     a     b
1     c     d
```
thanks!

**Reply by Mause** [reply]
> but this can only read json file?
> how to read a json str, I use python requests get the json str.
> thanks!

If you're using requests, you can use its built-in support for json, then wrap with a pandas DataFrame and insert from that using a replacement scan. I think that's presently the easiest way to load an array as rows from an inmemory string in python?

If you just want a single blob of json from an in memory string, you can pass the string as a parameter and cast `::JSON`

**Reply by ataft** [reply]
I'm looking for the same feature, but with the Node.js API. I have a JSON object (array with rows of data) that I'd like to write to a table, but currently I have to create a JSON file and then load from that.
